rotkehlchen/api/v1/fields.py

```diff
@@ -1029,3 +1029,32 @@ def _deserialize(
             raise ValidationError('List cant be empty')
 
         return result
+
+
+class EvmCounterpartyField(fields.Field):
+    """
+    Checks if the value provided is among a set of valid counterparties provided.
+    The set of valid options can be dynamically populated by calling set_counterparties
+    """
+
+    def __init__(self, *, counterparties: set[str] | None = None, **kwargs: Any) -> None:
+        self.counterparties = counterparties
+        super().__init__(**kwargs)
+
+    def set_counterparties(self, counterparties: set[str]) -> None:
+        self.counterparties = counterparties
+
+    def _deserialize(
+            self,
+            value: str,
+            attr: str | None,  # pylint: disable=unused-argument
+            data: Mapping[str, Any] | None,
+            **_kwargs: Any,
+    ) -> str:
+        if self.counterparties is None:
```

Comment:
Why is this needed? Can't you just make it required at input? This can only be a developer error. At the most it should be an assertion if anything